docker/build-ubuntu.ps1

```diff
@@ -3,31 +3,75 @@ param(
     $ArtifactsDir = "..\artifacts",
     $RavenDockerSettingsPath = "..\src\Raven.Server\Properties\Settings\settings.docker.posix.json",
     $Arch = "x64",
-    $DockerfileDir = "./ravendb-ubuntu")
+    $DockerfileDir = "./ravendb-ubuntu",
+    $DebPackagePath="",
+    [switch]$NoCache)
 
 $ErrorActionPreference = "Stop"
 
 . ".\common.ps1"
 
-function BuildUbuntuDockerImage ($version, $arch) {
+function GetPackageFileName($arch)
+{
     switch ($arch) {
         "arm32v7" { 
-            $packageFileName = "RavenDB-$version-raspberry-pi.tar.bz2"
-            break;
+            return "RavenDB-$version-raspberry-pi.tar.bz2"
         }
         "arm64v8" {
-            $packageFileName = "RavenDB-$version-linux-arm64.tar.bz2"
-            break;
+            return "RavenDB-$version-linux-arm64.tar.bz2"
         }
         "x64" {
-    $packageFileName = "RavenDB-$version-linux-x64.tar.bz2"
-            break;
+            return "RavenDB-$version-linux-x64.tar.bz2"
         }
         Default {
             throw "Arch not supported (currently x64 and arm32v7 are supported)"
         }
     }
+}
+
+function GetUbuntuVersionFromDockerfile($DockerfileDir, $DockerfileName) {
+    $dockerfilePath = Join-Path $DockerfileDir "Dockerfile.$arch"
+    $ubuntuVersion = Get-Content $dockerfilePath | Select-String -Pattern "(?<=FROM\s+mcr\.microsoft\.com\/dotnet\/runtime-deps:7\.0-)(.*)"
+    return $ubuntuVersion.Matches.Groups[1].Value
+}
+
+function SetupDebBuildEnvironment($arch, $ubuntuVersion){
+    switch ($arch) {
+        "x64" {
+            . "..\scripts\linux\pkg\deb\set-raven-platform-amd64.ps1"
+        }
+        "arm64v8" {
+            . "..\scripts\linux\pkg\deb\set-raven-platform-arm64.ps1"
+        }
+        "arm32v7" {
+            . "..\scripts\linux\pkg\deb\set-raven-platform-armhf.ps1"
+        }
+        Default {
+            Write-Error "ERROR: Unsupported architecture $($arch)"
+            exit 1
+        }
+    }
+
+    switch ($ubuntuVersion) {
+        "bionic" {
+            . "..\scripts\linux\pkg\deb\set-ubuntu-bionic.ps1"
+        }
+        "focal" {
+            . "..\scripts\linux\pkg\deb\set-ubuntu-focal.ps1"
+        }
+        "jammy" {
+            . "..\scripts\linux\pkg\deb\set-ubuntu-jammy.ps1"
+        }
+        Default {
+            Write-Error "ERROR: Unsupported Ubuntu version $($ubuntuVersion). Supported version: bionic, focal, jammy."
```

Comment:
In Powershell you can throw this instead of Write-Error. This will also include the stack trace that's going to help debugging.